hsmd/hsm.c

```diff
@@ -460,36 +463,52 @@ static void bitcoin_keypair(struct privkey *privkey,
 			      "BIP32 pubkey %u create failed", index);
 }
 
-static void create_new_hsm(void)
+static const char *hsm_filename(const tal_t *ctx,
+				const char *network_name)
+{
+	return tal_fmt(ctx, "hsm_secret-%s", network_name);
+}
+
+static int open_hsm_secret(const char *filename, int flags)
+{
+#ifdef COMPAT_V052
+	/* Filename used not to include network name */
+	rename("hsm_secret", filename);
```

Comment:
I suppose in practice it will not be likely, but in theory it would be possible for somebody running on testnet currently (or mainnet), stop, then update past this commit, then run on mainnet (or testnet), then fail.

In particular `wallet_network_check` comes after `hsm_init`, so the above mistake is possible to make `hsm_secret` of current network to be renamed incorrectly to other network.  It is recoverable by renaming the `hsm_secret` appropriately, but care must be done by those upgrading!